File: tp_3_design_pattern/Python/modeles/cours.py
# cours.py - represents a course (unite d'enseignement)


class Cours:
    """A course with a name and the professor who teaches it."""

    def __init__(self, nom_cours: str, professeur: str):
        self.nom_cours = nom_cours
        self.professeur = professeur

    def __str__(self) -> str:
        return f"{self.nom_cours} (Prof: {self.professeur})"

File: tp_3_design_pattern/Java/modeles/Cours.java
package modeles;

// a course with its name and the professor who teaches it
public class Cours {

    private String nomCours;
    private String professeur;

    public Cours(String nomCours, String professeur) {
        this.nomCours = nomCours;
        this.professeur = professeur;
    }

    public String getNomCours() {
        return nomCours;
    }

    public String getProfesseur() {
        return professeur;
    }

    @Override
    public String toString() {
        return nomCours + " (Prof: " + professeur + ")";
    }
}
